src/language/fromSourceInstaller.ts

```diff
@@ -104,9 +104,20 @@ export class FromSourceInstaller {
     // This works around this edge case.
     const injectPath = `PATH=${path.dirname(dotnet)}:$PATH`;
 
-    const projectToBuild = configuredVersionToNumeric(configuredVersion) < configuredVersionToNumeric('3.10.0')
-      ? 'DafnyLanguageServer/DafnyLanguageServer.csproj'
-      : 'DafnyDriver/DafnyDriver.csproj';
+    // If java is not installed, remove from Source/DafnyRuntime/DafnyRuntime.csproj
+    // the section <Target Name="BuildDafnyRuntimeJar">...</Target>
+    try {
+      await this.execLog(`${injectPath} java -version`);
+    } catch(e: unknown) {
+      // Read the project file, and then remove the section
+      let DafnyRunTimeCsprojContent = await fs.promises.readFile(Utils.joinPath(installationPath, 'dafny', 'Source', 'DafnyRuntime', 'DafnyRuntime.csproj').fsPath, 'utf8');
```

Comment:
Maybe this whole block should be in a try/catch as well, so that if it fails we still try to build.

Reply:
I suggest we simply update the script as needed when the time will come. Since it's a fallback script, we don't have a clear case anyway